CGEN Error unwrapping option interface field

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.9 b7b7a2a, press to see full `v doctor` output</summary>

|V full version      |V 0.4.9 45fd7eb.b7b7a2a
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 Pro 26100 64-bit
|Processor           |24 cpus, 64bit, little endian, AMD Ryzen 9 5900X 12-Core Processor  
|Memory              |72.58GB/127.92GB
|                    |
|V executable        |S:\repo\vlang\v.exe
|V last modified time|2025-01-19 20:55:33
|                    |
|V home dir          |OK, value: S:\repo\vlang
|VMODULES            |OK, value: C:\Users\john3\.vmodules
|VTMP                |OK, value: C:\Users\john3\AppData\Local\Temp\v_0
|Current working dir |OK, value: S:\vSelenium
|                    |
|Git version         |git version 2.45.1.windows.1
|V git status        |weekly.2025.1-123-gb7b7a2a9 (11 commit(s) behind V master)
|.git/config present |true
|                    |
|cc version          |N/A
|gcc version         |gcc (MinGW-W64 x86_64-ucrt-posix-seh, built by Brecht Sanders, r2) 14.2.0
|clang version       |(built by Brecht Sanders, r2) clang version 19.1.1
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |thirdparty-windows-amd64 b425ac82
|emcc version        |N/A
|glibc version       |N/A

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg -cg main.v && S:\vSelenium\main.exe`
```v
module main

import webdriver.edge { EdgeDriver, EdgeOptions }

fn open_browser(download_path string) ?EdgeDriver {
	mut options := EdgeOptions.init()
	options.add_experimental_option('excludeSwitches', ['enable-logging'])
	options.add_experimental_option('detach', true)
	options.add_experimental_option('prefs', {
		'download.default_directory': download_path
	})

	mut driver := EdgeDriver.init(options, none)

	return driver
}

fn main() {
	driver := open_browser('S:/Business Projects/PartsUnlimited/downloads')
	if driver == none {
		println('No Driver was created')
	} else {
		println('Successful driver creation')
	}
}

```

**What did you see?**
```
C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:805: warning: WINVER redefined

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:1207: warning: InterlockedIncrement64 redefined

In file included from C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:1572:

s:/repo/vlang/thirdparty/tcc/include/winapi/synchapi.h:13: warning: CreateEvent redefined

In file included from C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:1572:

s:/repo/vlang/thirdparty/tcc/include/winapi/synchapi.h:117: warning: OpenEvent redefined

In file included from C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:1572:

s:/repo/vlang/thirdparty/tcc/include/winapi/synchapi.h:167: warning: CreateMutex redefined

In file included from C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:1739:

S:/repo/vlang/thirdparty/vschannel/vschannel.c:284: warning: implicit declaration of function 'WSAConnectByNameW'

In file included from C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:1739:

S:/repo/vlang/thirdparty/vschannel/vschannel.c:300: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:11196: warning: implicit declaration of function 'tcc_backtrace'

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:17682: warning: implicit declaration of function 'GetFinalPathNameByHandleW'

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19186: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19199: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19216: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19223: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19261: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19271: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19344: warning: implicit declaration of function 'inet_ntop'

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19344: warning: cast between pointer and integer of different size

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:19354: warning: cast between pointer and integer of different size

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:20083: warning: assignment from incompatible pointer type

C:/Users/john3/AppData/Local/Temp/v_0/main.01JJ2V9WJC72C9MF0G8GMPNXRY.tmp.c:23459: error: expected struct or union but not 'struct _option_string *'
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

Clean compile



> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4013,6 +4013,9 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 						if i == 0 && (is_option_unwrap || nested_unwrap) {
 							deref := if g.inside_selector {
 								'*'.repeat(field.smartcasts.last().nr_muls() + 1)
+							} else if sym.kind == .interface && !typ.is_ptr()
+								&& field.orig_type.has_flag(.option) {
+								''
 							} else {
 								'*'
 							}
@@ -4196,7 +4199,8 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 	}
 	g.write(field_name)
 	if is_option_unwrap {
-		if field_typ.is_ptr() && g.table.final_sym(node.expr_type).kind in [.sum_type, .interface] {
+		if (is_dereferenced || field_typ.is_ptr())
+			&& g.table.final_sym(node.expr_type).kind in [.sum_type, .interface] {
 			g.write('->')
 		} else {
 			g.write('.')

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4199,8 +4199,7 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 	}
 	g.write(field_name)
 	if is_option_unwrap {
-		if (is_dereferenced || field_typ.is_ptr())
-			&& g.table.final_sym(node.expr_type).kind in [.sum_type, .interface] {
+		if g.table.final_sym(node.expr_type).kind in [.sum_type, .interface] {
 			g.write('->')
 		} else {
 			g.write('.')

--- a/vlib/v/tests/options/option_unwrap_selector_test.v
+++ b/vlib/v/tests/options/option_unwrap_selector_test.v
@@ -1,0 +1,22 @@
+interface IBar {
+	opt ?string
+}
+
+struct Bar implements IBar {
+	opt ?string
+}
+
+struct Foo {
+	field IBar
+}
+
+fn test_main() {
+	a := Foo{
+		field: Bar{
+			opt: 'foo'
+		}
+	}
+	if a.field.opt != none {
+		assert a.field.opt == 'foo'
+	}
+}

--- a/vlib/v/tests/options/option_unwrap_selector_test.v
+++ b/vlib/v/tests/options/option_unwrap_selector_test.v
@@ -10,6 +10,12 @@ struct Foo {
 	field IBar
 }
 
+fn (f &Foo) t() {
+	if f.field.opt != none {
+		assert f.field.opt == 'foo'
+	}
+}
+
 fn test_main() {
 	a := Foo{
 		field: Bar{
@@ -19,4 +25,5 @@ fn test_main() {
 	if a.field.opt != none {
 		assert a.field.opt == 'foo'
 	}
+	a.t()
 }

